packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -251,7 +251,8 @@ const Amount = ({
     <BaseBox
       paddingLeft="spacing.2"
       paddingRight="spacing.2"
-      display="flex"
+      // @TODO: fix casting of platform types. currently they all become `never` type
+      display={isReactNative ? castNativeType('flex') : castWebType('inline-flex')}
```

Comment:
So this is incorrect usage of castNativeType and castWebType function. It should ideally be `castWebType('flex' as BrandType)` but still figuring out what is the best and reusable way of doing this for such props. For now I am letting it be here and we can change that all at once later since this is how I have done it in other places too.

Reply:
Just cast it to never manually. no need to castNative function indirections. 

```
(isReactNative ? 'flex' : 'inline-flex') as never
```

Ref: #1234

Reply:
yeah casting only works in platform specific files(.native or .web) 😢

Reply:
> yeah casting only works in platform specific files(.native or .web)

So kamlesh, in this particular case this isn't working because the string `inline-flex` is not a branded type, thus castNativeType will cause it to become `never (AKA not possible type)` more [info here](https://github.com/razorpay/blade/pull/1229#discussion_r1211133318)

Reply:
Created it here - https://github.com/razorpay/blade/issues/1267